src/scan-imports.ts

```diff
@@ -77,48 +86,83 @@ function parseWebModuleSpecifier(specifier: string): null | string {
 }
 
 function getInstallTargetsForFile(filePath: string, code: string): InstallTarget[] {
+  const [imports] = parse(code) || [];
   const allImports: InstallTarget[] = [];
-  try {
-    const ast = parse(code, {plugins: ['dynamicImport'], sourceType: 'module'});
-    traverse(ast, {
-      ImportDeclaration(path) {
-        const webModuleSpecifier = parseWebModuleSpecifier(path.node.source.value);
-        if (webModuleSpecifier) {
-          allImports.push({
-            specifier: webModuleSpecifier,
-            all: false,
-            default: path.node.specifiers.some(s => s.type === 'ImportDefaultSpecifier'),
-            namespace: path.node.specifiers.some(s => s.type === 'ImportNamespaceSpecifier'),
-            named: path.node.specifiers
-              .map(s => s.type === 'ImportSpecifier' && s.imported.name)
-              .filter(Boolean),
-          });
-        }
-      },
-      Import(path) {
-        // Only match dynamic imports that are called as a function
-        if (path.parent.type !== 'CallExpression') {
-          return;
-        }
-        // Only match dynamic imports called with a single string argument
-        const [argNode] = path.parent.arguments;
-        if (argNode.type !== 'StringLiteral') {
-          return;
-        }
-        // Analyze that string argument as an import specifier
-        const webModuleSpecifier = parseWebModuleSpecifier(argNode.value);
-        if (webModuleSpecifier) {
-          allImports.push(createInstallTarget(webModuleSpecifier, true));
-        }
-      },
+  
+  for (const {s: start, e: end, ss: sStart, se: sEnd, d: dynamicImportIndex} of imports) {
+    const webModuleSpecifier = parseWebModuleSpecifier(code.substring(start, end));
+    if (!webModuleSpecifier) {
+      continue;
+    }
+    
+    const importStatement = code.substring(sStart, sEnd);
+    const dynamicImport = dynamicImportIndex > -1;
+    const defaultImport = !dynamicImport && !importStatement.includes('{');
```

Comment:
Wasn't totally sure on this as I didn't know if we wanted `default` to be true if the import statement includes both default and named imports, i.e. `import React, {useEffect} from ...` - at the moment this will be returned as `false`.

Reply:
I think @FredKSchott has more context on this, but I would guess this should be `default: true`? I interpret `import React, {useEffect}` as `import React …; const { useEffect } = React`. In other words, `React` is still the default export and contains all references; you’re just assigning a reference to one thing. But not sure if specifying `default: true` here has other implications.

Reply:
Actually on second thought 🤔 theoretically `useEffect` doesn’t have to be within the `default` export (I know it is here, but in another lib it wouldn’t have to be). Disregard my comment; I’m not sure! Need Fred’s opinion.

Reply:
Yea, we use these to tell the tree-shaker which imports we need from the final install target. So for `import React, {useEffect} from ...`:

- `{default: true}` - we need the default export (correct)
- `{default: false}` - we don't need the default export (incorrect)

Reply:
Which is a shame, because `importStatemen t.includes('{')` was a really clever way to check for this! 

You still may be able to do something like that with Regex though, by checking for non-whitespace characters between "import" and "from" (and outside of any `{}` block).